### PR TITLE
Add metrics for async secondary lock cleanup

### PR DESCRIPTION
**Pull request by wentaoxu**
1. count current async worker;
2. count elapse time for secondary lock cleanup task;
3. stat the result of secondary lock cleanup task;

### CODE REVIEW

**Comment by wentaoxu** [comment]
@disksing @zhangjinpeng1987 @siddontang PTAL

File: store/tikv/metrics.go [hunk 1]
```diff
@@ -172,6 +172,31 @@ var (
 			Name:      "load_safepoint_total",
 			Help:      "Counter of load safepoint.",
 		}, []string{"type"})
+
+	secondaryLockCleanupWorkerCounter = prometheus.NewGauge(
```
Comment (disksing) [inline store/tikv/metrics.go]: How about naming it _xxxGauge_ instead of _xxxCounter_?

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -570,10 +580,19 @@ func (c *twoPhaseCommitter) execute(ctx goctx.Context) error {
 		c.mu.RUnlock()
 		if !committed && !undetermined {
 			twoPhaseCommitGP.Go(func() {
+				start := time.Now()
+				secondaryLockCleanupWorkerCounter.Inc()
+				defer func() {
+					secondaryLockCleanupWorkerCounter.Dec()
```
Comment (disksing) [inline store/tikv/2pc.go]: Any need to use label values to distinguish commit and cleanup?

Comment (wentaoxu) [reply store/tikv/2pc.go]: Yes, I think it's better!

File: store/tikv/metrics.go (hunk 1 above)
Comment (wentaoxu) [reply store/tikv/metrics.go]: ok

**Comment by wentaoxu** [comment]
PTAL @disksing

File: store/tikv/2pc.go [hunk 3]
```diff
@@ -224,9 +224,19 @@ func (c *twoPhaseCommitter) doActionOnKeys(bo *Backoffer, action twoPhaseCommitA
 	if action == actionCommit {
 		// Commit secondary batches in background goroutine to reduce latency.
 		twoPhaseCommitGP.Go(func() {
+			start := time.Now()
+			secondaryLockCleanupWorkerGauge.Inc()
+			defer func() {
+				secondaryLockCleanupWorkerGauge.Dec()
+				secondaryLockCleanupHistogram.WithLabelValues("secondary_lock_cleanup").Observe(time.Since(start).Seconds())
```
Comment (siddontang) [inline store/tikv/2pc.go]: I think the tag name is duplicated with the metric name here.

Comment (siddontang) [reply store/tikv/2pc.go]: seem you don't need to use a MetricVec here, only a Metric is enough.

Comment (wentaoxu) [reply store/tikv/2pc.go]: I think I'd better to make a distinction between commit and rollback here.

**Comment by wentaoxu** [comment]
PTAL @siddontang

**Comment by wentaoxu** [comment]
PTAL @siddontang

File: store/tikv/2pc.go [hunk 4]
```diff
@@ -224,9 +224,19 @@ func (c *twoPhaseCommitter) doActionOnKeys(bo *Backoffer, action twoPhaseCommitA
 	if action == actionCommit {
 		// Commit secondary batches in background goroutine to reduce latency.
 		twoPhaseCommitGP.Go(func() {
+			start := time.Now()
+			secondaryLockCleanupWorkerGauge.Inc()
+			defer func() {
+				secondaryLockCleanupWorkerGauge.Dec()
+				secondaryLockCleanupHistogram.WithLabelValues("commit_cleanup_locks").Observe(time.Since(start).Seconds())
```
Comment (siddontang) [inline store/tikv/2pc.go]: using `commit` as the tag name is enough.

Comment (wentaoxu) [reply store/tikv/2pc.go]: ok

**Comment by wentaoxu** [comment]
PTAL @siddontang

**Comment by wentaoxu** [comment]
PTAL @AndreMouche

File: store/tikv/2pc.go [hunk 5]
```diff
@@ -224,9 +224,19 @@ func (c *twoPhaseCommitter) doActionOnKeys(bo *Backoffer, action twoPhaseCommitA
...
+				secondaryLockCleanupWorkerGauge.Dec()
+				secondaryLockCleanupHistogram.WithLabelValues("commit").Observe(time.Since(start).Seconds())
+			}()
+
 			e := c.doActionOnBatches(bo, action, batches)
 			if e != nil {
 				log.Debugf("2PC async doActionOnBatches %s err: %v", action, e)
+				secondaryLockCleanupTaskCounter.WithLabelValues("commit", "fail").Inc()
+			} else {
+				secondaryLockCleanupTaskCounter.WithLabelValues("commit", "ok").Inc()
```
Comment (zhangjinpeng1987) [inline store/tikv/2pc.go]: I think we just need record on fail condition.

Comment (wentaoxu) [reply store/tikv/2pc.go]: Maybe we want to know the percent of the failure, sometimes it's more useful than a single number.

Comment (wentaoxu) [reply store/tikv/2pc.go]: ok, though it has a global lock, I think it is better to remove.

**Comment by wentaoxu** [comment]
PTAL @zhangjinpeng1987

File: store/tikv/metrics.go [hunk 6]
```diff
@@ -172,6 +172,14 @@ var (
 			Name:      "load_safepoint_total",
 			Help:      "Counter of load safepoint.",
 		}, []string{"type"})
+
+	secondaryLockCleanupTaskCounter = prometheus.NewCounterVec(
```
Comment (zhangjinpeng1987) [inline store/tikv/metrics.go]: s/secondaryLockCleanupTaskCounter/secondaryLockCleanupFailedCounter

Comment (wentaoxu) [reply store/tikv/metrics.go]: Maybe `secondaryLockCleanupFailureCounter` is better?

**Comment by wentaoxu** [comment]
PTAL @zhangjinpeng1987

File: store/tikv/metrics.go [hunk 7]
```diff
@@ -172,6 +172,14 @@ var (
...
 			Help:      "Counter of load safepoint.",
 		}, []string{"type"})
+
+	secondaryLockCleanupFailureCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "tikvclient",
+			Name:      "lock_cleanup_task_total",
+			Help:      "failure statistic of secondary lock cleanup task.",
+		}, []string{"type", "result"})
```
Comment (zhangjinpeng1987) [inline store/tikv/metrics.go]: Only supply type is enough.

**Comment by zhangjinpeng1987** [comment]
@AndreMouche PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**